### PR TITLE
Add data structure and tags for starting on FD

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/DgSubcell/Tags/InitialTciData.hpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  using temporal_id = int;
+  using type = std::map<
+      temporal_id,
+      FixedHashMap<maximum_number_of_neighbors(Dim),
+                   std::pair<Direction<Dim>, ElementId<Dim>>,
+                   evolution::dg::subcell::InitialTciData,
+                   boost::hash<std::pair<Direction<Dim>, ElementId<Dim>>>>>;
+
+  template <typename ReceiveDataType>
+  static void insert_into_inbox(const gsl::not_null<type*> inbox,
```
Comment (kidder) [inline src/Evolution/DgSubcell/Tags/InitialTciData.hpp:34]: include for not_null

File: src/Evolution/DgSubcell/Tags/InitialTciData.hpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/functional/hash.hpp>
+#include <cstddef>
+#include <map>
+#include <optional>
```
Comment (kidder) [inline src/Evolution/DgSubcell/Tags/InitialTciData.hpp]: unused

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash these

**Comment by nilsdeppe** [comment]
Done. Thanks for the review!

**Review by kidder — APPROVED** [review]

**Pull request merged**